frame/staking/src/pallet/mod.rs

```diff
@@ -551,6 +604,80 @@ pub mod pallet {
 	#[pallet::getter(fn current_planned_session)]
 	pub type CurrentPlannedSession<T> = StorageValue<_, SessionIndex, ValueQuery>;
 
+	/// Wrapper struct for `ErasRewards`, `ClaimedRewards`, `ErasRewardsClipped`,
+	/// `ErasRewardsPaged`, `ErasTotalStake`.
+	pub(crate) struct EraInfo<T>(sp_std::marker::PhantomData<T>);
+	impl<T: Config> EraInfo<T> {
+		// TODO(ank4n): doc and clean up in 84 eras
+		// looks at ledger for older non paged rewards, and `ClaimedRewards` for newer paged
+		// rewards.
+		pub(crate) fn temp_is_rewards_claimed(
+			era: EraIndex,
+			ledger: &StakingLedger<T>,
+			validator: &T::AccountId,
+			page: PageIndex,
+		) -> bool {
+			ledger.claimed_rewards.binary_search(&era).is_ok() &&
+				Self::is_rewards_claimed(era, validator, page)
+		}
+
+		fn is_rewards_claimed(era: EraIndex, validator: &T::AccountId, page: PageIndex) -> bool {
+			ClaimedRewards::<T>::get(era, validator).iter().find(|&&p| page == p).is_some()
+		}
+
+		pub(crate) fn get_validator_exposure(
+			era: EraIndex,
+			validator: &T::AccountId,
+			page: PageIndex,
+		) -> Exposure<T::AccountId, BalanceOf<T>> {
+			return match <ErasStakersPaged<T>>::get((era, validator, page)) {
+				Some(paged_exposure) => paged_exposure,
+				// only return clipped exposure if page zero and no paged exposure entry
+				None if page == 0 => <ErasStakersClipped<T>>::get(&era, &ledger.stash),
+				_ => Default::Default(),
+			}
+		}
+
+		pub(crate) fn set_rewards_as_claimed(
+			era: EraIndex,
+			validator: &T::AccountId,
+			page: PageIndex,
+		) {
+			ClaimedRewards::<T>::mutate(era, validator, |pages| {
+				pages.push(page);
+			})
+		}
+
+		pub(crate) fn set_validator_exposure(
+			era: EraIndex,
+			validator: &T::AccountId,
+			exposure: Exposure<T::AccountId, BalanceOf<T>>,
+		) {
+			<ErasStakers<T>>::insert(era, &validator, &exposure);
+
+			exposure
+				.in_chunks_of(T::ExposurePageSize::get() as usize)
+				.iter()
+				.enumerate()
+				.for_each(|(page, paged_exposure)| {
+					<ErasStakersPaged<T>>::insert((era, &validator, page as u32), &paged_exposure);
+				});
+
+			// fixme(ank4n): no need to keep storing new clipped exposure
+			let mut exposure_clipped = exposure;
+			let clipped_max_len = T::MaxNominatorRewardedPerValidator::get() as usize;
+			if exposure_clipped.others.len() > clipped_max_len {
+				exposure_clipped.others.sort_by(|a, b| a.value.cmp(&b.value).reverse());
+				exposure_clipped.others.truncate(clipped_max_len);
+			}
+			<ErasStakersClipped<T>>::insert(era, &validator, exposure_clipped);
```

Comment:
We can get rid of storing clipped exposure. For new eras (once this PR is in production), we only need paged exposure for rewards payout and it also gives a nice separation of paged and non paged exposures, that is
all non-paged exposure are in clipped, and we are slowly migrating to paged exposures for new eras. The `get_validator_exposure `  abstracts to the caller what `exposure` to return.